thunder/tests/opinfos.py

```diff
@@ -2749,6 +2749,64 @@ def tril_sample_generator(op, device, dtype, requires_grad, **kwargs):
 # data_movement_ops.append(convert_element_type_opinfo)
 
 
+def type_sample_generator_tensor(op, device, dtype, requires_grad, **kwargs):
```

Comment:
Take a look at the sample generator for `to`

https://github.com/k223kim/lightning-thunder/blob/2430d477ed6bf70e3cfd58aa52dffdb21d89fb95/thunder/tests/opinfos.py#L2810

This will currently just check that a tensor can be cast to its current dtype, but it'd be interesting to verify it can be upcast (to double or complex double), too